clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -281,8 +281,9 @@ public void testCommitAsyncWithNullCallback() {
         final ArgumentCaptor<AsyncCommitEvent> commitEventCaptor = ArgumentCaptor.forClass(AsyncCommitEvent.class);
         verify(applicationEventHandler).add(commitEventCaptor.capture());
         final AsyncCommitEvent commitEvent = commitEventCaptor.getValue();
-        assertEquals(offsets, commitEvent.offsets());
-        assertDoesNotThrow(() -> commitEvent.future().complete(null));
+        assertTrue(commitEvent.offsets().isPresent());
+        assertEquals(offsets, commitEvent.offsets().get());
+        assertDoesNotThrow(() -> commitEvent.future().complete(offsets));
```

Comment:
I know this line is in the original code, but I have a question: in what case would calling `CompletableFuture.complete()` possibly throw an exception?

Reply:
Agree, we should not test Java implementation which is not part of Kafka code. Remove `assertDoesNotThrow` for `CompletableFuture.complete()`. Thank you.